### PR TITLE
Automate Dead Link Detection in Docs

### PR DESCRIPTION
**Pull request by fjlanasa**
### Pull Request Description
Closes https://github.com/alteryx/evalml/issues/3364

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

File: .github/workflows/broken_link_checker.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  my-broken-link-checker:
+    name: Check for broken links (${{ matrix.page }})
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        page:
+          - index
+          - install
+          - start
+          - tutorials
```
Comment (freddyaboulton) [inline .github/workflows/broken_link_checker.yml:17]: Also, I think we should fix the existing broken links found by this check in this PR!

File: .github/workflows/broken_link_checker.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  schedule:
+      - cron: '0 7 * * *'
+
+jobs:
+  my-broken-link-checker:
+    name: Check for broken links (${{ matrix.page }})
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        page:
```
[the remark this answers is not in the document]
Comment (fjlanasa) [reply .github/workflows/broken_link_checker.yml:13]: i am not sure if there is a good way to do that without also recursively crawling all links within each page.

[the remark this answers is not in the document]
Comment (freddyaboulton) [reply .github/workflows/broken_link_checker.yml:13]: Does that lead to many false positives?

File: .github/workflows/broken_link_checker.yml (hunk 1 above)
Comment (fjlanasa) [reply .github/workflows/broken_link_checker.yml:17]: Good point. I only found one in my testing, but forgot to push up the fix. Will do.

File: .github/workflows/broken_link_checker.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (fjlanasa) [reply .github/workflows/broken_link_checker.yml:13]: I can give it a shot. The main issue is that it takes a very long time, but I initially ruled it out when I thought we would run this check for each PR, so I guess it's not as big of an issue if it still works.

**Pull request closed without merging**